Assets/Scripts/Puzzles/Color/Line.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DN.Puzzle.Color
+{
+	public class Line : MonoBehaviour
+	{
+		public LineColor LineColor => lineColor;
+		public Node StartingNode => startingNode;
+		public Node EndNode => endNode;
+		public bool CanTraverseBothWays => canTraverseBothWays;
+
+		[SerializeField] private LineColor lineColor;
+		[SerializeField] private Node startingNode;
+		[SerializeField] private Node endNode;
+		[SerializeField] private bool canTraverseBothWays;
+		[SerializeField] private Image lineSprite;
+
+		private GameObject arrow;
+
+		private void Awake()
+		{
+			if (!startingNode || !endNode || !lineSprite)
+			{
+				Debug.LogError("Assign all elements first.", gameObject);
+				return;
+			}
+
+			arrow = Resources.Load<GameObject>("Line Pointer");
+
+			UnityEngine.Color color = SetColor();
+			float rotation = SetRotation();
+			SetLength();
+			CreateArrow(rotation, color);
+		}
+
+		private UnityEngine.Color SetColor()
+		{
+			ColorPuzzleSettings settings = Resources.Load<ColorPuzzleSettings>("ColorPuzzleSettings");
+			lineSprite.color = settings.ColorSettings.First(x => x.LineColor == lineColor).Color;
+			return lineSprite.color;
+		}
+
+		private float SetRotation()
+		{
+			Vector3 vector = endNode.transform.position - startingNode.transform.position;
+			float rotation = Mathf.Atan2(
+				vector.y,
+				vector.x) * Mathf.Rad2Deg;
+			rotation = rotation < 0 ? rotation + 360f : rotation;
+			transform.eulerAngles = new Vector3(transform.eulerAngles.x, transform.eulerAngles.y, rotation);
+			return rotation;
+		}
+
+		private float SetLength()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/manual/stepperman/pvb-programming/pullRequest?prid=5403921)